### ISSUE TITLE
Change the way of monetizing

### ISSUE THREAD

**Issue by rradar** [routerkeygen/routerkeygenAndroid#47]
Just as an idea (and for compatibility with google free phones)

Maybe it would be possible to change the way of financing from the integrated adds (I don't know how much is earned with this but I doubt it makes anybody rich besides the advertising industrie!) to financing based on **donations** and/or platforms like **patreon** or maybe even a little **kickstarter** for the future development.

I think the downside's for the users with a app with adds are bigger (annoyance, tracking, security) than the financial benefit for the repo owner.

Maybe it would be possible to have an extra branch for this version and keep on with the add filled version on google play?

**Comment by ph4r05** [comment]
Agree with you about the pain from ads.
On the other hand, there are also ads-free forks of this nice project - for a minority of users concerned with the privacy and related problems connected to ads.

I can see the reasons the Ads is current monetization alternative of choice. IMO Majority of users are OK with this and those can significantly contribute to income.

**Comment by rradar** [comment]
@ph4r05 
You have any link to a up to date repo were the apk's are released? Thank's a lot!

EDIT:
I found this nice repo here: https://github.com/yolosec/routerkeygenAndroid/
But the latest released apk is still a 3.x release and not the 4.1.5 from the code :-(

A life without google is a hard to live!

**Comment by ph4r05** [comment]
Ah you cannot use google play to install packages, hmm. Do you need an APK file? 
EDIT: APK release added

**Comment by rradar** [comment]
Great! Thank's a lot @ph4r05 

So this issue is closed. This fork should satisfy anybody https://github.com/yolosec/routerkeygenAndroid/

**Comment by pidiet** [comment]
Sorry to bother. I have problem with linked  APK. Does not run on Android 7.1 without gapps.

Anyway to get it running? Sorry, issues are closed at yolosec :(

**Comment by rradar** [comment]
I can confirm. Also the release from yolosec (@ph4r05) doesn't work without google-apps. Only way to use this app right now is to go with a old release (3.x).

Would be nice to get the modifications integrated to make the 4.x releases to work without google apps.

**Comment by ph4r05** [comment]
Hmm the problem might be related to a new permission model and `com.google.android.gms:play-services-location` package required to access location-related information.
EDIT: I can try to make a version without gapps needed. Issues @yolosec are enabled again.

**Comment by rradar** [comment]
![dfg_2018-03-01-14-18-32](https://user-images.githubusercontent.com/34582688/36846931-e06e79e2-1d5c-11e8-967b-bde089057b3c.png)
This message comes when starting the app (so there is some check if the gapps are installed)

![dfg_2018-03-01-14-18-28](https://user-images.githubusercontent.com/34582688/36846967-010a5acc-1d5d-11e8-8fbd-c5df562ed7e9.png)
Here it never finds a network (screen stays empty after scanning)

![dfg_2018-03-01-14-18-20](https://user-images.githubusercontent.com/34582688/36846982-12451d72-1d5d-11e8-9bc5-56e95248e525.png)
And all requested permissions are granted

**Comment by ph4r05** [comment]
Thanks for the report!
@rradar @pidiet Please give a try the following app flavor without google play services

https://github.com/yolosec/routerkeygenAndroid/releases/download/v4.1.5/routerKeygen-noGapps-release.apk

Please let me know if that works for you. Thanks!

**Comment by rradar** [comment]
Thank's so much for this special blend @ph4r05 

I did a quick run and here are my findings:
- Message that gapps are missing is gone

![dfg_2018-03-01-15-22-33](https://user-images.githubusercontent.com/34582688/36849603-c0f1c7b0-1d64-11e8-8e4c-a0299efdb25b.png)
- Now it's reported that the app doesn't find a network (even though there is at least 5 of them)

If I am right it should list all available networks and cluster them for known weaknesses.

I don't know what would be needed to dig deeper? If you need a log file you can point me to it and I would like to provide it!

Thank's again for your efforts! I could not do anything of this!

**Comment by ph4r05** [comment]
@rradar thanks for testing this!
Hmmm I tried it in the emulator with google-play-free image and it worked there. Android 7.0, API 24. 
Can you pls go to the system settings, find the app and check whether it has also location privileges?  And what is the Android version you are using? 

Btw - logcat would be also nice, to check whether there is some exception or something.
Debugging build
https://github.com/yolosec/routerkeygenAndroid/releases/download/v4.1.5/routerKeygen-noGapps-debug.apk

Thanks!

**Comment by rradar** [comment]
Hey @ph4r05 
Android is 7.1.2 (lineageos based) and privileges are granted (location & storage). I just reinstalled  it again and granted the privileges one more time. Same output - no results.

**Comment by ph4r05** [comment]
@rradar unfortunately, in the lightweight fork there is no log reporting mechanism.
It would be great if you possibly could extract logs from your device. Here are some tips:

- https://developer.android.com/studio/debug/bug-report.html
- https://android.gadgethacks.com/how-to/android-basics-capture-logcat-for-detailed-bug-reports-0167774/

**Comment by rradar** [comment]
Hey @ph4r05 

I tried the noGapps version (https://github.com/yolosec/routerkeygenAndroid/releases/download/v4.1.5/routerKeygen-noGapps-release.apk) on another device (Android 5.1.1) and this time **it worked fearless**! I also tried it on another Android 7.1.2 and it **also show's all wifi networks**. There seams to be a local problem with my phone.

So thank's a lot for all your effort and work! I saw you put the noGapps also in the release section from github! Very nice! Hopefully many people will find it! Once again: Thank you so much! Have a very nice weekend ;)
